hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketDeleteRequest.java

```diff
@@ -198,6 +215,35 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     }
   }
 
+  private boolean bucketContainSnapshot(OMMetadataManager omMetadataManager,
+      String snapshotBucketKey) throws IOException {
+    // check both in cache and table
+    Iterator<Map.Entry<CacheKey<String>, CacheValue<SnapshotInfo>>> cacheIter =
+        omMetadataManager.getSnapshotInfoTable().cacheIterator();
+    while (cacheIter.hasNext()) {
+      Map.Entry<CacheKey<String>, CacheValue<SnapshotInfo>> cacheKeyValue =
+          cacheIter.next();
+      String key = cacheKeyValue.getKey().getCacheKey();
+      if (key.startsWith(snapshotBucketKey)) {
+        return true;
+      }
+    }
+    try (TableIterator<String, ? extends Table.KeyValue<String, SnapshotInfo>>
+        snapshotIterator = omMetadataManager
+            .getSnapshotInfoTable().iterator()) {
+      snapshotIterator.seek(snapshotBucketKey);
+      if (snapshotIterator.hasNext()) {
+        Table.KeyValue<String, SnapshotInfo> dbKeyValue =
+            snapshotIterator.next();
+        String dbKey = dbKeyValue.getKey();
+        if (dbKey.startsWith(snapshotBucketKey)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
```

Comment:
This should do it atm. Though I wonder if we should have a more efficient way of checking whether a bucket has snapshots (and how many snapshots the bucket has -- which can be utilized by Recon) later.

Reply:
I think we could even use the Snapshot Chain feature that was added earlier so every time we don't  look up the DB.  I will check if we can use it somehow for this purpose . One thing that needs to be done is while delete snapshot is implemented it has to update the in-memory chain and the DB both.

Reply:
Just checked the code , looks like `SnapshotChainManager` is not used anywhere in OM code. I think we can use it to list snapshots / to check if snapshots exist. What do you think? 
cc @neils-dev